lib/log_parser.rb

```diff
@@ -1,23 +1,38 @@
 require "json"
+
 class LogParser
   def initialize(file_path)
     @file_path = file_path
     raise "File not found" unless File.exists?(@file_path)
-    @file = File.open(@file_path)
-    @file_data = @file.readlines.map(&:chomp)
-    @file.close
+    file = File.open(@file_path)
+    @file_data = file.readlines.map(&:chomp)
+    file.close
   end
 
   def get_first_line
     first_line = @file_data[0]
   end
 
   def get_log_data
-    data = {:lines => file_length}
+    players = []
+    @file_data.each do |line|
+      if line.match(/ \d{1,2}:\d{1,2} Kill: \d+ \d+ \d+: .+ killed .+by .+/)
+        player_one = line[line.rindex(":")+2..line.index("killed")-2]
+        player_two = line[line.index("killed")+7..line.rindex("by")-2]
+        players.push(player_one)
+        players.push(player_two)
+      end
+      if line.match(/ \d{1,2}:\d{1,2} ClientUserinfoChanged: \d+ n\\.+\\t\\\d+\\model/)
+        player = line[line.index("n\\")+2..line.index("\\t")-1]
+        players.push(player)
+      end
+    end
```

Comment:
```suggestion
    players = @file_data.flat_map do |line|
      if line.match(/ \d{1,2}:\d{1,2} Kill: \d+ \d+ \d+: .+ killed .+by .+/)
        player_one = line[line.rindex(":")+2..line.index("killed")-2]
        player_two = line[line.index("killed")+7..line.rindex("by")-2]
        
        [player_one, player_two]
      end
      
      if line.match(/ \d{1,2}:\d{1,2} ClientUserinfoChanged: \d+ n\\.+\\t\\\d+\\model/)
        line[line.index("n\\")+2..line.index("\\t")-1]
      end
    end
```

Explicando, um `.map` irá passar por cada linha e simplesmente retornar uma linha modificada baseada na original. Um flat_map é exatamente a mesma coisa que um `.map.flatten`, vou dar um exemplo simplificado com números:

```ruby
irb(main):010:0> [1, 2, 3].map{|n| n * 10}
=> [10, 20, 30]

irb(main):011:0> [1, 2, 3].map{|n| [n, n * 10] }
=> [[1, 10], [2, 20], [3, 30]]

irb(main):012:0> [1, 2, 3].map{|n| [n, n * 10] }.flatten
=> [1, 10, 2, 20, 3, 30]

irb(main):013:0> [1, 2, 3].flat_map{|n| [n, n * 10] }
=> [1, 10, 2, 20, 3, 30]

irb(main):014:0> [1, 2, 3].flat_map{|n| n.odd? ? n : [] }
=> [1, 3]
```